frontend/src/metabase/query_builder/components/view/ViewHeader.jsx

```diff
@@ -460,43 +461,51 @@ function ViewTitleHeaderRightSide(props) {
         />
       )}
       {QuestionNotebookButton.shouldRender(props) && (
-        <QuestionNotebookButton
-          ml={2}
-          question={question}
-          isShowingNotebook={isShowingNotebook}
-          setQueryBuilderMode={setQueryBuilderMode}
-          data-metabase-event={
-            isShowingNotebook
-              ? `Notebook Mode;Go to View Mode`
-              : `View Mode; Go to Notebook Mode`
-          }
-        />
+        <ViewHeaderIconButtonContainer>
+          <QuestionNotebookButton
+            iconSize={16}
+            ml={2}
+            question={question}
+            isShowingNotebook={isShowingNotebook}
+            setQueryBuilderMode={setQueryBuilderMode}
+            data-metabase-event={
+              isShowingNotebook
+                ? `Notebook Mode;Go to View Mode`
+                : `View Mode; Go to Notebook Mode`
+            }
+          />
+        </ViewHeaderIconButtonContainer>
       )}
       {NativeQueryButton.shouldRender(props) && (
-        <NativeQueryButton
-          size={16}
-          question={question}
-          data-metabase-event="Notebook Mode; Convert to SQL Click"
-        />
+        <ViewHeaderIconButtonContainer>
+          <NativeQueryButton
+            size={16}
+            question={question}
+            data-metabase-event="Notebook Mode; Convert to SQL Click"
+          />
+        </ViewHeaderIconButtonContainer>
       )}
       {hasExploreResultsLink && <ExploreResultsLink question={question} />}
       {hasRunButton && (
-        <RunButtonWithTooltip
-          className={cx("text-brand-hover text-dark", {
-            hide: isShowingNotebook,
-            "text-white-hover": isResultDirty,
-          })}
-          medium
-          borderless
-          ml={1}
-          compact
-          result={result}
-          isRunning={isRunning}
-          isDirty={isResultDirty}
-          isPreviewing={isPreviewing}
-          onRun={() => runQuestionQuery({ ignoreCache: true })}
-          onCancel={cancelQuery}
-        />
+        <ViewHeaderIconButtonContainer>
+          <RunButtonWithTooltip
+            className={cx("text-brand-hover text-dark", {
+              hide: isShowingNotebook,
+              "text-white-hover": isResultDirty,
+            })}
+            iconSize={16}
+            onlyIcon
```

Comment:
change from `borderless` to `onlyIcon` as `borderless` still render border but with `transparent` color causing the size difference when changing between active and inactive button.

https://github.com/metabase/metabase/blob/13de0b8327c1250e74fd321234f216b502ae5726/frontend/src/metabase/css/components/buttons.css#L110-L114
vs
https://github.com/metabase/metabase/blob/13de0b8327c1250e74fd321234f216b502ae5726/frontend/src/metabase/css/components/buttons.css#L120-L125